content/storage-history/history.textile

```diff
@@ -0,0 +1,633 @@
+    ```
+
+    ```[python]
+    rest = AblyRest('{{API_KEY}}')
+    channel = rest.channels.get('{{RANDOM_CHANNEL_NAME}}')
+    result_page = channel.history()
+    recent_message = result_page.items[0]
+    print('Most recent message: ' + str(recent_message.message.id) + ' - ' + recent_message.data)
+    ```
+
+    ```[php]
+    $rest = new Ably\AblyRest('{{API_KEY}}');
+    $channel = $rest->channels->get('{{RANDOM_CHANNEL_NAME}}');
+    $channel->publish('example', 'message data');
+    $resultPage = $channel->history();
+    $recentMessage = $resultPage->items[0];
+    echo("Most recent message data: " . $recentMessage->data);
+    ```
+
+    ```[java]
+    AblyRest rest = new AblyRest("{{API_KEY}}");
+    Channel channel = rest.channels.get("{{RANDOM_CHANNEL_NAME}}");
+    channel.publish("example", "message data")
+    PaginatedResult<Message> resultPage = channel.history(null);
+    Message recentMessage = resultPage.items[0];
+    System.out.println("Most recent message: " + recentMessage.id + " - " + recentMessage.data);
+    ```
+
+    ```[csharp]
+    AblyRest rest = new AblyRest("{{API_KEY}}");
+    IRestChannel channel = rest.Channels.Get("{{RANDOM_CHANNEL_NAME}}");
+    await channel.PublishAsync("example", "message data");
+    PaginatedResult<Message> resultPage = await channel.HistoryAsync();
+    Message recentMessage = resultPage.Items[0];
+    Console.WriteLine("Most recent message: " + recentMessage.Id + " - " + recentMessage.Data);
+    ```
+
+    ```[objc]
+    ARTRest *rest = [[ARTRest alloc] initWithKey:@"{{API_KEY}}"];
+    ARTRestChannel *channel = [rest.channels get:@"{{RANDOM_CHANNEL_NAME}}"];
+    [channel publish:@"example" data:@"message data"];
+    [channel history:^(ARTPaginatedResult<ARTMessage *> *resultPage, ARTErrorInfo *error) {
+        ARTMessage *recentMessage = resultPage.items[0];
+        NSLog(@"Most recent message: %@ - %@", recentMessage.id, recentMessage.data);
+    }];
+    ```
+
+    ```[swift]
+    let rest = ARTRest(key: "{{API_KEY}}")
+    let channel = rest.channels.get("{{RANDOM_CHANNEL_NAME}}")
+    channel.publish("example", data: "message data")
+    channel.history { resultPage, error in
+        let recentMessage = resultPage!.items[0] as! ARTMessage
+        print("Most recent message: \(recentMessage.id) - \(recentMessage.data)")
+    }
+    ```
+
+    ```[go]
+    rest, err := ably.NewRestClient(ably.NewClientOptions("{{API_KEY}}"))
+    channel := rest.Channels.Get("{{RANDOM_CHANNEL_NAME}}")
+    err = channel.Publish("example", "message data")
+    page, err := channel.History(nil)
+    fmt.Println("Last published message: %s\n", page.Messages[0].Data)
+
+On success, the returned @PaginatedResult@ encapsulates an array of Message objects corresponding to the current page of results. @PaginatedResult@ supports pagination using next and first methods.
+
+Failure to retrieve the message history will raise an @AblyException@.
```

Comment:
This looks like info from the API reference? Not sure we need to be explaining the callback here, as it'll differ for every language.